doc/rados/operations/crush-map-edits.rst

```diff
@@ -240,121 +247,136 @@ and two node buckets named ``host`` and ``rack`` respectively.
+       after bulk purchases).  When storage devices have exactly the same
+       weight, you may use the ``uniform`` bucket type, which allows CRUSH to
+       map replicas into uniform buckets in constant time. If your devices have
+       non-uniform weights, you should not use the uniform bucket algorithm.
+
+    #. **list**: List buckets aggregate their content as linked lists. The 
+       behavior of list buckets is governed by the :abbr:`RUSH (Replication
+       Under Scalable Hashing)`:sub:`P` algorithm. In the behavior of this
+       bucket type, an object is either relocated to the newest device in
+       accordance with an appropriate probability, or it remains on the older
+       devices as before. This results in optimal data migration when items are
+       added to the bucket. The removal of items from the middle or the tail of
+       the list, however, can result in a significant amount of unnecessary
+       data movement. This means that list buckets are most suitable for
+       circumstances in which they **never shrink or very rarely shrink**.
+
+    #. **tree**: Tree buckets use a binary search tree. They are more efficient
+       at dealing with buckets that contain many items than are list buckets.
+       The behavior of tree buckets is governed by the :abbr:`RUSH (Replication
+       Under Scalable Hashing)`:sub:`R` algorithm.  Tree buckets reduce the
+       placement time to 0(log\ :sub:`n`). This means that tree buckets are
+       suitable for managing large sets of devices or nested buckets.
+
+    #. **straw**: Straw buckets allow all items in the bucket to "compete" 
+       against each other for replica placement through an aleatory process
+       analogous to drawing straws. This is different from the behavior of list
+       buckets and tree buckets, which use a divide-and-conquer strategy that
+       either gives certain items precedence (for example, those at the
+       beginning of a list) or obviates the need to consider entire subtrees of
+       items. Such an approach improves the performance of the replica
+       placement process, but can also introduce suboptimal reorganization
+       behavior when the contents of a bucket change due an addition, a
+       removal, or the re-weighting of an item.
+
+        * **straw2**: Straw2 buckets improve on Straw by correctly avoiding 
+          any data movement between items when neighbor weights change.  For
+          example, if the weight of a given item changes (including during the
+          operations of adding it to the cluster or removing it from the
+          cluster), there will be data movement to or from only that item.
+          Neighbor weights are not taken into account.
+
 
 .. topic:: Hash
 
-   Each bucket uses a hash algorithm. Currently, Ceph supports ``rjenkins1``.
-   Enter ``0`` as your hash setting to select ``rjenkins1``.
+   Each bucket uses a hash algorithm. Ceph supports the ``rjenkins1``
```

Comment:
Ceph today supports only the

Reply:
I'll make the change, but why the emphasis on "today"?

Reply:
I went with "as of Reef" instead of "today", because the today on which your comment was written isn't even the today on which I'm writing this note back to you.

Reply:
If there has even been another hash algorithm, it was before my time, and I suspect there has never been another.  Might be another in the future though.  nbd